unit_tests/CMakeLists.txt

```diff
@@ -18,7 +18,16 @@ FetchContent_Declare(
         GIT_TAG        release-${GOOGLETEST_VERSION}
 )
 
-FetchContent_MakeAvailable(googletest)
+if (${CMAKE_VERSION} VERSION_LESS "3.14")
+  message("Cmake < 3.14 - consider upgrading (FetchContent_MakeAvailable())")
+  FetchContent_GetProperties(googletest)
+  if(NOT googletest_POPULATED)
```

Comment:
I'm not 100% sure this will work since CMake is a bit weird with versions. because we require CMake 3.12, newer versions may downgrade themselves to 3.12 when executing the script. It's simpler just to use the add_subdirectory approach since it works on newer versions of CMake and it's basically what MakeAvailable does just a bit more verbose